kolibri/core/logger/csv_export.py

```diff
@@ -144,5 +149,6 @@ def csv_file_generator(facility, log_type, filepath, overwrite=False):
         for item in queryset.select_related("user", "user__facility").values(
             *log_info["db_columns"]
         ):
-            writer.writerow(map_object(item))
-            yield
+            if (item["start_timestamp"]) >= start and (item["start_timestamp"]) <= end:
```

Comment:
I think that the time filters should be applied to the queryset in https://github.com/learningequality/kolibri/blob/develop/kolibri/core/logger/csv_export.py#L85 & https://github.com/learningequality/kolibri/blob/develop/kolibri/core/logger/csv_export.py#L100 instead of  here, so only the needed data is fetched from the database. Appling the date filters here would mean all the logs are retrieved and filter afterwards.

Reply:
Yes, I understand what you mean about fetching only what is needed. So I should apply the date filters as an addition to the filtering that is happening [here](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/logger/csv_export.py#L100:~:text=queryset%20%3D%20log_info%5B%22queryset%22%5D.filter(dataset_id%3Dfacility.dataset_id))?

>     queryset = log_info["queryset"].filter(dataset_id=facility.dataset_id)

Reply:
yep, either there or in the lines I've commented above. Actually, your suggestion is a better place as its just one place insted of two. _Luckily_ the name of the fields are the same in both loggers (`start_timestamp` & `end_timestamp` )